product/views/ManageIQ_Providers_InfraManager_Template.yaml

```diff
@@ -26,6 +26,8 @@ cols:
 - allocated_disk_storage
 - last_scan_on
 - region_description
+- product_name
```

Comment:
I was chatting with Kavya about this one.  This is just a method on vm or template.  It isn't a virtual column and doesn't preload the operation_systems table so it could be an N+1.

Some alternatives:
1) Use the include section of the [report yaml](https://github.com/ManageIQ/manageiq/blob/master/product/reports/100_Configuration%20Management%20-%20Virtual%20Machines/059_Guest%20OS%20Information%20(any%20OS).yaml#L16-L34) like:
```
include:
  operating_system:
    columns:
    - product_name
```

and use:

```
col_order:
- operating_system.product_name
```

The downside is this method on a vm has logic that looks for the product name in other places if operating_systems doesn't have it.

2) Make this method a virtual column with proper uses operating_system.
Like [os_image_name](https://github.com/ManageIQ/manageiq/blob/3b4f5050af8a90515f066927d63d239da69be481/app/models/vm_or_template.rb#L142), see [also](https://github.com/ManageIQ/manageiq/blob/3b4f5050af8a90515f066927d63d239da69be481/app/models/vm_or_template.rb#L558-L565).
3) ???

Reply:
Note, I don't know if there are pros/cons of those approaches.  I vaguely recall that `include` section in the report yaml came first and likely gave the benefits of the virtual column uses before we had that.  But, I can't tell if there's a reason to use one or the other now.

Reply:
Maybe @NickLaMuro @kbrock remember re: virtual column with uses vs. report yaml include section.

Reply:
I can maybe spend some time with this code later today and see what the differences are, but off the top of my head, I don't really have an answer.

Feel free to bug me on this one if I don't get back to you on this today.

Reply:
I didn't get to this Friday, but making a plan on looking at it today.

That said, if I find something this is problematic, I can always update it in a follow up PR.

Reply:
Yeah, I think we can always make `product_name` a virtual column with uses to avoid N+1 issues or replace this with `include` of `operating_system` and use `operating_system.product_name` but if we want consistency with what `product_name` does in terms of handling nil `operating_sytem` and nil `operating_system.product_name`, it would make sense to do the former.